src/audio/host.c

```diff
@@ -359,33 +359,24 @@ static uint32_t host_get_copy_bytes_normal(struct comp_dev *dev)
 		/* limit bytes per copy to one period for the whole pipeline
 		 * in order to avoid high load spike
 		 */
-		const uint32_t free_bytes =
-			audio_stream_get_free_bytes(&hd->local_buffer->stream);
+		free_bytes = audio_stream_get_free_bytes(&hd->local_buffer->stream);
 		copy_bytes = MIN(hd->period_bytes, MIN(avail_bytes, free_bytes));
+		if (!copy_bytes)
+			comp_info(dev, "no bytes to copy, %d free in buffer, %d available in DMA",
+				  free_bytes, avail_bytes);
```

Comment:
checking the copy_bytes before ALIGN_DOWN actually changes the logging logic a little bit here, e.g. if 0 < copy_bytes < dma_copy_align, we will not log "no bytes to copy" anymore while we do previously.